src/injected/textSelectorEngine.ts

```diff
@@ -48,17 +48,27 @@ export function createTextSelector(shadow: boolean): SelectorEngine {
   return engine;
 }
 
+const SINGLE_QUOTE_CHAR = "'";
+const DOUBLE_QUOTE_CHAR = '"';
+const isSurroundedBy = (text: string, check: string) => text.length > 2 && text[0] === check && text[text.length - 1] === check;
+export const hasTextSelectorSurroundings = (selector: string) => isSurroundedBy(selector, DOUBLE_QUOTE_CHAR) || isSurroundedBy(selector, SINGLE_QUOTE_CHAR);
+
 type Matcher = (text: string) => boolean;
 function createMatcher(selector: string): Matcher {
-  if (selector[0] === '"' && selector[selector.length - 1] === '"') {
-    const parsed = JSON.parse(selector);
+  // If the selector is surrounded by quotes test case sensitive
+  if (hasTextSelectorSurroundings(selector)) {
+    const innerText = selector.slice(1, selector.length - 1);
+    // Use JSON.parse since we want to parse escaped characters
+    const parsed = innerText.includes('\\') ? JSON.parse(`"${innerText}"`) : innerText;
```

Comment:
This does not work for `text='foo"bar'` because it will try to parse `"foo"bar"`. I guess we have to do our own parser, or forget about single quotes. WDYT?

Reply:
From my side to leave out the single quote support would be not that good. Too often there are code bases / users out there which would run into that issue.

I'm just thinking, why at all do we want to allow inside the text selector JSON based input?

A few test cases which I've tested and come to the conclusion, that we might not need `JSON.parse`:

- `text="foo"yes"very'tricky"` - would work since we do only test directly for the text inside the quotes
- `text=foo\\backslash` - would test for `foo\backslash`
- `text=""ya"` - would test for `"ya"`

I see only the `text="\\"ya"` case in the [tests](https://github.com/mxschmitt/playwright/blob/feature/enh-text-selector-engine/test/queryselector.spec.js#L516) which makes JSON parse useful. But since we are adding single quote support and dont use JSON parse anymore, they can either do `text=""ya"` or `text='"ya'` to handle that specific use case which should also be very rare.

Reply:
The main reason is to follow standard practices with strings. If you quote a string, you should escape the quote inside, so that we can figure out where does the string end. For example, consider `text="foo">>text="bar"`: is it a single text selector with `foo">>text="bar` body, or two chained text selectors?

Reply:
hmm, good point. Should it be allowed to have multiple text selectors in a single selector?

If yes, then we probably have to write our own parser. If not, I see not a problem, that `text="foo">>text="bar"` will lead to `foo">>text="bar`.

Maybe in the future we should also allow `#foo > text="foo"`, but that would be out of the scope of this PR.